pkg/agent/agent.go

```diff
@@ -900,47 +900,60 @@ func (i *Initializer) initNodeLocalConfig() error {
 		return nil
 	}
 
-	// Parse all PodCIDRs first, so that we can support IPv4/IPv6 dual-stack configurations.
-	if node.Spec.PodCIDRs != nil {
-		for _, podCIDR := range node.Spec.PodCIDRs {
-			_, localSubnet, err := net.ParseCIDR(podCIDR)
-			if err != nil {
-				klog.ErrorS(err, "Failed to parse subnet from Pod CIDR string", "CIDR", podCIDR)
-				return err
-			}
-			if localSubnet.IP.To4() != nil {
-				if i.nodeConfig.PodIPv4CIDR != nil {
-					klog.InfoS("One IPv4 PodCIDR is already configured on this Node, ignoring the IPv4 Subnet CIDR", "subnet", localSubnet)
+	err = wait.PollImmediate(10*time.Second, 30*time.Second, func() (bool, error) {
+		node, err = i.client.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
```

Comment:
could we move the poll to the head of `initNodeLocalConfig` to avoid a duplicate `Node.Get` query at L803?

Reply:
We could. But then, we will redo also the processing of the other config parameters. Or we should check if these were already initialized and reprocessed due to a failure in PodCIDRs processing. The code could become more cumbersome IMO.
Is it worth it just to save a call to `Node.Get`?

Reply:
> But then, we will redo also the processing of the other config parameters.

Will it? I meant code like this:
```
var node *corev1.Node
var podIPv4CIDR, podIPv6CIDR *net.IPNet
if err := wait.PollImmediate(5*time.Second, 30*time.Second, func() (bool, error) {
        // Code that checks PodCIDRs
        podIPv4CIDR = ...
        podIPv6CIDR = ...
}); err != nil {
	klog.ErrorS(err, "Spec.PodCIDR is empty for Node. Please make sure --allocate-node-cidrs is enabled "+
		"for kube-controller-manager and --cluster-cidr specifies a sufficient CIDR range", "nodeName", nodeName)
	return fmt.Errorf("CIDR string is empty for Node %s", nodeName)
}

// Code that processes the other config parameters

// nodeInterface is the interface that has K8s Node IP. transportInterface is the interface that is used for
// tunneling or routing the traffic across Nodes. It defaults to nodeInterface and can be overridden by the
// configuration parameters TransportInterface and TransportInterfaceCIDRs.
var nodeInterface, transportInterface *net.Interface
...
```
It just ensures the Node is valid first, then does real initialization. I think it also avoids unnecessary initialization when the Node is invalid.